<a href="https://colab.research.google.com/github/hritikchaturvedi11/MLnow_2.0/blob/main/Univeriate_linear_regression_prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**Dataset**: There are total of _26 labels_ in the dataset. Three mostly advertised features-> _Carbody(Sedan,Hatchback), engine size_ and _peakRPM_ is related to the price is already available in the dataset. We will be only looking the _**engine size**_ 

[Link to Download the dataset](https://www.kaggle.com/hellbuoy/car-price-prediction)

Problem statement: Predict the car price from it's engine size



**Date Created**: June , 2021

Topic: Univariate Linear Regression

**Author**:
 Hritik Chaturvedi

[[Assessment Repository](https://github.com/hritikchaturvedi11/MLnow_2.0)]

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
cd '/content/drive/MyDrive/Machine Learning google colab/EDA/Car_Price_Prediction'

/content/drive/MyDrive/Machine Learning google colab/EDA/Car_Price_Prediction


In [3]:
ls

CarPrice_Assignment.csv


Importing the libraries

In [5]:
import pandas as pd
import numpy as np
import tensorflow as tf
import os
from tensorflow import keras

Importing the data

In [8]:
priceData = pd.read_csv('CarPrice_Assignment.csv');

Let's know more about the data.

In [9]:
priceData.head()

car_ID  symboling                   CarName  ... citympg highwaympg    price
0       1          3        alfa-romero giulia  ...      21         27  13495.0
1       2          3       alfa-romero stelvio  ...      21         27  16500.0
2       3          1  alfa-romero Quadrifoglio  ...      19         26  16500.0
3       4          2               audi 100 ls  ...      24         30  13950.0
4       5          2                audi 100ls  ...      18         22  17450.0

[5 rows x 26 columns]

In [11]:
priceData.shape

(205, 26)

Now we know we have 205 price  details and 26 labels. But for our today's problem we need to drop out the features that are not necessary.